javaparser-core/src/main/java/com/github/javaparser/ast/StubUnit.java

```diff
@@ -0,0 +1,153 @@
+package com.github.javaparser.ast;
+
+import com.github.javaparser.TokenRange;
+import com.github.javaparser.ast.visitor.GenericVisitor;
+import com.github.javaparser.ast.visitor.VoidVisitor;
+import com.github.javaparser.metamodel.InternalProperty;
+import com.github.javaparser.printer.PrettyPrinter;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.github.javaparser.Providers.UTF8;
+
+/**
+ * <p>
+ * This class represents a stub file. The stub file is a Java file, but with the optionally omitted
```

Comment:
A stub file isn't just a Java file with some information omitted.  It's a concatenation of multiple Java files.
Without this information, "the list of compilation units" below is confusing.